### ISSUE TITLE
Cannot access RuntimeContext in open()

### ISSUE THREAD

**Issue by AHeise** [stratosphere/stratosphere#244]
Invoking getRuntimeContext() in open() inside a RegularPactTask always causes an IllegalStateException. 

In RegularPactTask initLocalStrategies() and thus open() of the stub is called before the RuntimeContext is set with initialize().

Please note that inside a ChainedMapTask it works as expected

Here is the corresponding fragment of RegularPactTask:

``` java
// initialize the remaining data structures on the input and trigger the local processing
// the local processing includes building the dams / caches

try {
    int numInputs = driver.getNumberOfInputs();

    initInputsSerializersAndComparators(numInputs);
    initLocalStrategies(numInputs);

} catch (Exception e) {
    throw new RuntimeException("Initializing the input processing failed" +

        e.getMessage() == null ? "." : ": " + e.getMessage(), e);

}

if (!this.running) {

    if (LOG.isDebugEnabled())
        LOG.info(formatLogString("Task cancelled before PACT code was started."));

    return;
}

// pre main-function initialization

initialize();
```

**Comment by StephanEwen** [comment]
Are you using a combiner that is executed as part of a local strategy (combining sort) ? That is the only point I can think of where the UDFs have cross-dependencies with the local strategies...

**Comment by AHeise** [comment]
Yes it definitively breaks in a combiner.

See https://s3.amazonaws.com/archive.travis-ci.org/jobs/13543000/log.txt (search for “runtime context has not been initialized”)

**Comment by StephanEwen** [comment]
Okay, should be a quick fix...

**Comment by StephanEwen** [comment]
Fixed in 36aabd4e9472659ed3320817e6f2ced4404397b8
